lib/coap/client.ex

```diff
@@ -69,4 +61,11 @@ defmodule CoAP.Client do
       @wait_timeout -> %Message{}
     end
   end
+
+  defp as_ip_tuple(ip) do
```

Comment:
This is required because the IP address as binary worked for `:gen_udp`, but did not match the connection id from peer ip+port+token.